misk-jobqueue/src/testFixtures/kotlin/misk/jobqueue/FakeJobQueue.kt

```diff
@@ -268,13 +290,34 @@ data class FakeJob(
   var deadLettered: Boolean = false
     internal set
 
+  var backoffDelayedTime: Instant? = null
+    internal set
+
+  var delayDuration: Long? = null
+    internal set
+
   override fun acknowledge() {
     acknowledged = true
   }
 
   override fun deadLetter() {
     deadLettered = true
   }
+  override fun backOffDelay() {
```

Comment:
i honestly don't like this implementation this much, so would be happy to hear ideas/thoughts on the alternatives!